mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -135,9 +135,131 @@ class ConvConverter final
   }
 };
 
+class BroadcastConverter final
+    : public OpConversionPattern<migraphx::BroadcastOp> {
+public:
+  using OpConversionPattern<migraphx::BroadcastOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(migraphx::BroadcastOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    auto operands = adaptor.getOperands();
+    Location loc = op->getLoc();
+    auto input_t = operands[0];
+    auto axis = op->getAttr("axis").cast<IntegerAttr>().getInt();
+
+    for (auto &use : op->getResult(0).getUses()) {
+      auto expandedOp = use.getOwner();
+      // isa binary operation,
+      if (isa<migraphx::AddOp>(expandedOp) ||
```

Comment:
Shouldn't need to restrict to these ops. Can we just insert replace the migraphx.broadcast with the tosa.reshape?

Reply:
broadcast in tosa is restrictive. Just checked and it only broadcasts on the second input of the binary operations. Not clearly defined as a rule but only defined per each operator.
I'll change it to reflect that rule.